patroni/ha.py

```diff
@@ -257,9 +271,15 @@ def manual_failover_process_no_leader(self):
         return self._is_healthiest_node(members, check_replication_lag=False)
 
     def is_healthiest_node(self):
+        if self.is_paused() and self.cluster.failover and not self.cluster.failover.scheduled_at:
```

Comment:
It can't be moved below 284, because it is a special case when we are doing manual failover in the paused state. It will not even reach the line 284, because either it will return from this method on the line 278 or 281.
But you are right about 'nofailover', this check also should be added here.